### PR TITLE
Getting /api endpoint based on arbitrary server address

### PR DESCRIPTION
**Pull request by TomaszKolek**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@TomaszKolek, it looks like you've already signed the Dropbox CLA. Thanks!

File: redmine_zulip/init.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
     description 'Sends notifications to Zulip.'
     version '0.91'
     url 'https://github.com/zulip/zulip-redmine-plugin'
-    author_url 'https://www.zulip.com/'
+    author_url 'https://www.zulipchat.com/'
```
Comment (timabbott) [inline redmine_zulip/init.rb]: lets make this https://www.zulip.org

File: redmine_zulip/lib/zulip_hooks.rb [hunk 2]
```diff
@@ -105,10 +105,7 @@ def zulip_port()
    end
 
    def zulip_api_basename()
-     if Setting.plugin_redmine_zulip[:zulip_server]["api.zulip.com"]
-       return ""
-     end
-     return "/api"
+     return (Setting.plugin_redmine_zulip[:zulip_server].sub %r{^https?:(//|\\\\)(www\.)?}i, '').concat('/api')
```
Comment (TomaszKolek) [inline redmine_zulip/lib/zulip_hooks.rb]: I saw that in all of our integrations we sent a request to "api.zulip.com" not to "https://api.zulip.com" not sure if its important but I'd like to escape the protocol part

Comment (timabbott) [inline redmine_zulip/lib/zulip_hooks.rb]: OK, makes sense; let's drop the `www` piece (I don't think anyone will be typing `www` URLs) but leave the http(s) stuff.

Comment (TomaszKolek) [inline redmine_zulip/lib/zulip_hooks.rb]: Done

**Comment by timabbott** [comment]
Cool, merged.  Just to double-check, you've tested this works, right?

**Comment by TomaszKolek** [comment]
@timabbott I just unit-tested the method.

**Pull request merged**